jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/InputStreamContentSource.java

```diff
@@ -59,17 +77,17 @@ public Content.Chunk read()
 
         try
         {
-            byte[] buffer = new byte[getBufferSize()];
-            int read = inputStream.read(buffer);
+            ByteBuffer buffer = bufferPool.acquire(getBufferSize(), false);
+            int read = inputStream.read(buffer.array(), buffer.arrayOffset(), buffer.capacity());
```

Comment:
```suggestion
            int read = inputStream.read(buffer.array(), buffer.arrayOffset(), buffer.remaining());
```
This is more correct. We have `MappedByteBufferPool.Tagged` that slices the buffers of the pool, so `remaining()` is correct, not `capacity()`.